### PR TITLE
extend init to accept rasr binaries

### PR DESCRIPTION
**Pull request by christophmluscher**
the idea is to set the rasr binaries at the beginning. at the moment this still uses `gs.RASR_*` but we could remove this. it would be cleaner but backward comp?!

### CODE REVIEW

**Comment by christophmluscher** [comment]
we might want to wait for merging until Nicks PR (in preparation) is merged first..

File: common/setups/rasr/rasr_system.py
Comment (christophmluscher) [reply common/setups/rasr/rasr_system.py:62]: I would remove `gs.RASR_*` completely

**Comment by michelwi** [comment]
> it would be cleaner but backward comp?!

yes, I also agree to remove it completely. If a user wants to use a specific RASR version from a certain path in the FS, then he could just give it as a Path object to this function.

But now the argument `rasr_root` should be required and not optional. and `rasr_arch` can default to "linux-x86_64-standard" and then you can remove the if-else-logic when you set it later.

File: common/setups/rasr/gmm_system.py
Comment (JackTemaki) [inline common/setups/rasr/gmm_system.py:93]:
```suggestion
        rasr_binary_path: Optional[tk.Path] = None,
```
We are not not passing the root, but a path to a folder directly containing the binaries, as they can be independent from the repository.

File: common/setups/rasr/gmm_system.py
Comment (JackTemaki) [inline common/setups/rasr/gmm_system.py:96]:
```suggestion
        super().__init__(rasr_binary_path=rasr_binary_path, rasr_arch=rasr_arch)
```

File: common/setups/rasr/rasr_system.py
Comment (JackTemaki) [inline common/setups/rasr/rasr_system.py:54]:
```suggestion
        rasr_binary_path: Optional[tk.Path] = None,
```

File: common/setups/rasr/rasr_system.py
Comment (JackTemaki) [inline common/setups/rasr/rasr_system.py:59]:
```suggestion
        self.rasr_binary_path = rasr_binary_path
```

File: common/setups/rasr/rasr_system.py
Comment (JackTemaki) [inline common/setups/rasr/rasr_system.py:113]:
```suggestion
        if self.rasr_binary_path is not None and self.rasr_arch is not None:
```

File: common/setups/rasr/rasr_system.py
Comment (JackTemaki) [inline common/setups/rasr/rasr_system.py:115]:
```suggestion
                rasr_root=self.rasr_binary_path, rasr_arch=self.rasr_arch
```

File: common/setups/rasr/rasr_system.py
Comment (JackTemaki) [inline common/setups/rasr/rasr_system.py:117]:
```suggestion
        elif self.rasr_binary_path is not None and self.rasr_arch is None:
```

File: common/setups/rasr/rasr_system.py
Comment (JackTemaki) [inline common/setups/rasr/rasr_system.py:118]:
```suggestion
            self.crp[corpus_key].set_executables(rasr_binary_path=self.rasr_binary_path)
```

**Review by JackTemaki — CHANGES_REQUESTED** [review]

File: common/setups/rasr/rasr_system.py
Comment (JackTemaki) [inline common/setups/rasr/rasr_system.py:115]:
```suggestion
                rasr_binary_path=self.rasr_binary_path, rasr_arch=self.rasr_arch
```

**Review by JackTemaki — APPROVED** [review]

**Comment by michelwi** [comment]
I still stand to my previous comment/suggestion:

> the argument rasr_root should be required and not optional. and rasr_arch can default to "linux-x86_64-standard" and then you can remove the if-else-logic when you set it later.

And I think a doc string would be helpful to avoid the confusion that chris/me originally had

> We are not not passing the root, but a path to a folder directly containing the binaries, as they can be independent from the repository.

But otherwise I am ok with the current state of the code.

File: common/setups/rasr/rasr_system.py
Comment (michelwi) [inline common/setups/rasr/rasr_system.py:124]:
```suggestion
        self.crp[corpus_key].set_executables(
            rasr_binary_path=self.rasr_binary_path, rasr_arch=self.rasr_arch
        )
```
If the defaults are not None any more, we don't need to test for it.

**Review by michelwi — APPROVED** [review]

**Pull request merged**
